### DISCUSSION TITLE
Add NetEase Cloud Music to the lyrics plugin as a source

### DISCUSSION THREAD

**Discussion by amogus07** [beetbox/beets#5347] [Ideas]
The one thing I always have to do before importing music to my library is manually add synchronized lyrics using Oto Music on my phone, which, I'm pretty sure, fetches lyrics from [NetEase Cloud Music](https://music.163.com) (I use Syncthing to synchronize files with my laptop). NetEase has lyrics for lots of songs where other providers don't have any, or only non-synchronized ones. That process is very tedious, so it would be awesome to have beets do it for me! I heard there is some geo-blocking going on with NetEase, but there do seem to be workarounds.

**Comment by amogus07** [comment]
There's an FOSS Android app ([SongSync](https://github.com/Lambada10/SongSync)) that implements it too, as well as other sources (Apple Music, LRCLib, Spotify (currently seems to be broken) and QQ Music).

**Comment by snejus** [comment]
I'd suggest to create an issue for this new feature. Do they have an API by any chance?

**Reply by amogus07** [reply]
I took a quick glance at the relevant part of the source code and SongSync seems to be using a third-party API: https://lyrics.paxsenix.org/

reference implementation: https://github.com/Lambada10/SongSync/tree/master/app/src/main/java/pl/lambada/songsync/data/remote/lyrics_providers

**Reply by amogus07** [reply]
https://github.com/pxeemo/lrxy

**Reply by amogus07** [reply]
took a closer look now, turns out SongSync only uses paxsenix for Apple Music ~and QQ~
